cmd/maya-exporter/app/collector/collector.go

```diff
@@ -15,25 +15,39 @@
 package collector
 
 import (
+	"sync"
+
 	v1 "github.com/openebs/maya/pkg/stats/v1alpha1"
 	"github.com/prometheus/client_golang/prometheus"
 	"k8s.io/klog"
 )
 
 // collector implements prometheus.Collector interface
 type collector struct {
+	sync.Mutex
+	request bool
 	Volume
 	metrics
 }
 
+func (c *collector) isRequestInProgress() bool {
```

Comment:
Better to add a comment here for caller to take a lock before calling this function.

Reply:
roger that

Reply:
Is it possible to add lock/unlock in this function and avoid the caller from having to worry about the locks?